### DISCUSSION TITLE
How to type/configure the response media type?

### DISCUSSION THREAD

**Discussion by stumpylog** [vitalik/django-ninja#1108] [Q&A] [answered]
I have some end points which return either a known content file `FileResponse` or a image content type.  Based on the [Swagger documentation](https://swagger.io/docs/specification/media-types/), it should be possible to include this information in the API.

Does anyone know how this could be done with Ninja?

**Comment by vitalik** [comment, marked as answer]
@stumpylog 


here is a reference to discussion - #1073

generally for how you can do the following:

```Python
BIN_RESPONSE = {
    "responses": {
        200: {
            "description": "OK",
            "content": {"application/octet-stream": {"schema": {"type": "string", "format": "binary"}}},
        }
    }
}


@api.get("/hello", openapi_extra=BIN_RESPONSE)
def hello(request):
    return HttpResponse(b'hello', content_type='application/octet-stream')
```

**Reply by stumpylog** [reply]
Perfect, thank you.  I checked discussion, not issues.  Whoops.
